control/veda_val.py

```diff
@@ -356,9 +356,12 @@ def send_200(self, val_api_return):
         VAL will not allow duped studio urls to be sent,
         so we must scrub the data
         """
-        for c in val_api_return['courses']:
-            if c in self.val_data['courses']:
-                self.val_data['courses'].remove(c)
+        for course in val_api_return['courses']:
+            for key, entry in course.items():
+                for crs in self.val_data['courses']:
+                    for ky, ey in crs.items():
+                        if ky == key:
+                            self.val_data['courses'].remove(crs)
```

Comment:
You can `break` after this line - you don't have to check any more keys in the course once you've removed the course.